apps/site/src/app/schedule/sections/SchedulePage.tsx

```diff
@@ -0,0 +1,140 @@
+"use client";
+
+import { useState, useEffect, useRef } from "react";
+import * as Tabs from "@radix-ui/react-tabs";
+
+import EventCard from "../components/EventCard";
+import convertToPST from "@/lib/utils/convertToPST";
+
+import "@radix-ui/themes/styles.css";
+import "./SchedulePage.scss";
+
+const T_REFRESH = 15000;
+
+const weekdayFormat = new Intl.DateTimeFormat("en", {
+	weekday: "long",
+});
+
+interface ScheduleProps {
+	schedule: {
+		title: string;
+		eventType: string;
+		location?: string | undefined;
+		virtual?: string | undefined;
+		startTime: Date;
+		endTime: Date;
+		organization?: string | undefined;
+		hosts?: string[] | undefined;
+		description: JSX.Element;
+	}[][];
```

Comment:
Thought: it seems we're repeating a lot of `EventProps`. Would there be a way to derive the necessary field types?

Reply:
Or better yet, would there be a way to use inference from zod, e.g. `z.infer<typeof Events>`?